src/controllers/user.controller.ts

```diff
@@ -0,0 +1,124 @@
+import * as Koa from 'koa'
+import {hashSync, compare} from 'bcrypt'
+import {getRepository, Repository} from 'typeorm'
+import User from '../models/user.entity'
+import * as HttpStatus from 'http-status-codes'
+import * as JWT from 'jsonwebtoken'
+import mailer from "../controllers/mail.controller";
+require('dotenv').config()
+
+export type MessageType ={
+    to:string,
+    subject: string,
+    html:string
+}
+
+export default class UserController {
+
+    static async getUsers(ctx: Koa.Context) {
+        const userRepo: Repository<User> = getRepository(User)
+        const users: User[] = await userRepo.find()
+        ctx.body = {
+            data: {users}
+        }
+    }
+
+    static async getUser(ctx: Koa.Context) {
+        const userRepo: Repository<User> = getRepository(User)
+        const user: User = await userRepo.findOne(ctx.params.user_id)
+        if (!user) {
+            ctx.throw(HttpStatus.NOT_FOUND)
+        }
+        ctx.body = {
+            data: {user}
+        }
+    }
+
+    static async registerUser(ctx: Koa.Context) {
+        const userRepo: Repository<User> = getRepository(User)
+        const checkUser: User = await userRepo.findOne({email: ctx.request.body.email})
+        if (checkUser) {
+            ctx.throw(HttpStatus.BAD_REQUEST, 'User already exists')
+        }
+        ctx.request.body.password = hashSync(ctx.request.body.password, 10)
+        const token: string | string[] = ctx.query.token
+        const message: MessageType= {
+            to: ctx.request.body.email,
+            subject: 'Welcome to our site!',
+            html: `<div>Поздравляем, Вы почти  зарегистрировались на нашем сайте! </div> 
+                <div>Перейдите по ссылке для подтерждения аккаунта </div>
+                <a href="${ctx.URL.href}/${token}"> Register me</a>`
```

Comment:
Yeah, so Vitaly already mentioned that it is better to use some temporary token just for confirmation than exposing the token with what you later can access the application.